examples/qualcomm/scripts/dummy_llama2.py

```diff
@@ -92,15 +93,17 @@ def create_device_inputs(example_inputs, use_kv_cache):
     pte_filename = "dummy_llama2_qnn"
 
     use_fp16 = False if args.ptq else True
-    build_executorch_binary(
-        instance.get_eager_model().eval(),
-        instance.get_example_inputs(),
-        args.model,
-        f"{args.artifact}/{pte_filename}",
-        inputs,
-        custom_annotations=(),
-        use_fp16=use_fp16,
-    )
+    # Temp bypass the dynamo tracing error, for kv-cache mode only
+    with torch.nn.attention.sdpa_kernel([SDPBackend.MATH]), torch.no_grad():
```

Comment:
Just some questions for it.
1. Do we have alternative ops for slice_scatter and select_scater?
2. `python3.10 dummy_llama2.py -m "SM8450" -b artifact/ -kv --ptq` has different results when using/not using DecomposeScaledDotProductAttention. Is it expected?

Thanks

Reply:
1. Yes we will. Ideally will be gone with the updated kv cache implementation
2. You mean like the graph is different now with this mode? What is the output graph with/without this pass?

Reply:
>Yes we will. 

Thanks a lot! That's awesome.

>You mean like the graph is different now with this mode? What is the output graph with/without this pass?

I think so. Based on this PR, the behavior of `python3.10 dummy_llama2.py -m "SM8450" -b artifact/ -kv --ptq` is different. [With this pass](https://github.com/pytorch/executorch/blob/f327e53b3bb176a9a0bceab1bb94d6d754acedc5/backends/qualcomm/quantizer/quantizer.py#L280) process is broken at [`annotated_model = prepare_pt2e(captured_model, quantizer)`](https://github.com/pytorch/executorch/blob/f327e53b3bb176a9a0bceab1bb94d6d754acedc5/examples/qualcomm/scripts/utils.py#L158). However it will keep going and break at [`exir_exported_program = exir.capture(...)`](https://github.com/pytorch/executorch/blob/f327e53b3bb176a9a0bceab1bb94d6d754acedc5/backends/qualcomm/utils/utils.py#L59) if we comment out this pass.